packages/amplify-graphql-relational-transformer/src/utils.ts

```diff
@@ -117,6 +144,26 @@ export function ensureFieldsArray(config: HasManyDirectiveConfiguration | HasOne
   } else if (config.fields.length === 0) {
     throw new InvalidDirectiveError(`No fields passed to @${config.directiveName} directive.`);
   }
+
+  if (config.references) {
+    throw new InvalidDirectiveError(`DynamoDB models do not support 'references' on @${config.directiveName} directive.`);
+  }
+}
+
+export function ensureReferencesArray(
+  config: HasManyDirectiveConfiguration | HasOneDirectiveConfiguration | BelongsToDirectiveConfiguration,
+) {
+  if (!config.references) {
+    throw new InvalidDirectiveError(`References must be passed to @${config.directiveName} directive for RDS models.`);
+  } else if (!Array.isArray(config.references)) {
+    config.references = [config.references];
+  } else if (config.references.length === 0) {
+    throw new InvalidDirectiveError(`No references passed to @${config.directiveName} directive.`);
+  }
+
+  if (config.fields) {
+    throw new InvalidDirectiveError(`Relational database models do not support 'fields' on @${config.directiveName} directive.`);
```

Comment:
same as above: This can be checked first so we can fail fast without running the above checks on `references`. We do have a schema validator package that runs similar schema validations, is it possible to consolidate these there?